packages/pwa/app/utils/responsive-image.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {getAppOrigin} from 'pwa-kit-react-sdk/utils/url'
+
+/**
+ * @param {Object} props
+ * @param {string} props.src - The image's url
+ * @param {Object} [props.sizes] - Defines how the image responds across Chakra breakpoints
+ * @param {number[]} [props.widths] - Will be used to generate the image's srcset
+ * @param {Object} [props.transformations] - How to transform the image's url {@link https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/content/b2c_commerce/topics/image_management/b2c_creating_image_transformation_urls.html}
+ * @return {Object} - Props for an image component
+ *
+ * @example
+ * <Img
+ *     {...getImageProps({
+ *         src,
+ *         sizes: {base: 'calc(100vw / 2)', md: '500px'},
+ *         widths: [500, 1000, 1500],
+ *         transformations: {q: 60}
+ *     })}
+ * />
+ */
+export const getImageProps = ({src, sizes, widths, transformations, ...otherProps}) => {
```

Comment:
Here the `sizes` and `widths` are optional because:
- For more flexibility, let's not assume that every image is gonna be responsive images.
- Also, even if they're for responsive images, developers typically won't input them until later in development (when they know more about their responsive design).